Fix the lambda() type definition

jontg opened the pull request:
`lambda()` [returns a function with the signature of handler](https://github.com/alexa-js/alexa-app/blob/master/index.js#L643-L646), it's signature does not directly match `handler()`.

jontg commented:
Of particular note to @lazerwalker and @dblock who appear to [have authored the original type definition](https://github.com/alexa-js/alexa-app/commit/e3b966d7191639eda27464a0c379dd21cb395738)... yesterday?

lazerwalker commented:
Ack, good catch! Looks good — mind updating the changelog quickly?

jontg commented:
Sure... I've got another fix in the pipeline because it looks like `request.getSession().application.applicationId` doesn't match the typing, but I'll do that separately

jontg commented:
@lazerwalker updated CHANGELOG in the amended commit

dblock commented on CHANGELOG.md:
```diff
@@ -12,6 +12,8 @@
 * [#250](https://github.com/alexa-js/alexa-app/pull/250): Add support for handling and/or delegating Dialog - [@sbonami](https://github.com/sbonami).
 * [#258](https://github.com/alexa-js/alexa-app/pull/258): Fix ability to edit request.type() in app.pre - [@samueljr](https://github.com/samueljr).
 * [#266](https://github.com/alexa-js/alexa-app/pull/266): Add TypeScript definitions - [@lazerwalker](https://github.com/lazerwalker).
+* [#269](https://github.com/alexa-js/alexa-app/pull/269): Fix the lambda() type definition - [@jontg](https://github.com/jontg).
```
4.1.0 was released, move this to 4.1.1

dblock commented on types/index.d.ts:
```diff
@@ -73,7 +73,7 @@ export class app {
 
   /** For backwards compatibility */
-  lambda: (event: string, context: alexa.Context, callback: (error: Error, response: response) => void) => void;
+  lambda: () => (event: string, context: alexa.Context, callback: (error: Error, response: response) => void) => void;
```
Feels like there should be a test for the regression. Possible, please?

dblock commented on CHANGELOG.md:
```diff
@@ -2,7 +2,7 @@
 
 ### 4.1.1 (Next)
 
-* Your contribution here
```
Put this back please.

dblock commented:
Just for my personal education, in what way (what's the error?) does using these two functions fail without the fix?

jontg commented:
@dblock I updated the CHANGEME; for the test failure:
```bash
$ git checkout master
Switched to branch 'master'
Your branch is up-to-date with 'origin/master'.
$ git checkout patch-1 -- types/test.ts
$ node_modules/.bin/dtslint types/
/Users/jontg/Development/alexa-app/types/test.ts
ERROR: 21:31  expect  TypeScript compile error: Supplied parameters do not match any signature of call target.```

dblock commented:
Needs a rebase with #271 which just got merged. Thanks.

dblock commented on CHANGELOG.md:
```diff
@@ -13,6 +14,7 @@
 * [#258](https://github.com/alexa-js/alexa-app/pull/258): Fix ability to edit request.type() in app.pre - [@samueljr](https://github.com/samueljr).
 * [#266](https://github.com/alexa-js/alexa-app/pull/266): Add TypeScript definitions - [@lazerwalker](https://github.com/lazerwalker).
 
+
```
Space is cheap, but still ;)

jontg replied:
hah, sorry fixing :-)

jontg commented:
@dblock all set, thanks for the reminder!

dblock commented:
Thanks for contributing @jontg, glad to see @lazerwalker's work on TypeScript stuff is already being useful, so thanks @lazerwalker for bringing it in in #266.

dblock commented:
Merged.

jontg commented:
Thanks again for your review and help @dblock and @lazerwalker !
